File: LeetCode-November-challenge/ReverseVowelsofaString.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
private:
    bool isVowel(char ch){
        return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' || ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U';
    }
public:
    string reverseVowels(string s) {
        int low = 0, high = s.size() - 1;
        while(low < high){
            if(isVowel(s[low]) && isVowel(s[high])){
                swap(s[low], s[high]);
                low++, high--;
            } else if(!isVowel(s[low])) low++;
            else high--;
        }
        return s;
    }
};
int main(){
    string s = "hello";
    Solution S;
    cout<<S.reverseVowels(s)<<endl;
    return 0;
}

File: LeetCode-November-challenge/ReverseVowelsofaString.js
var reverseVowels = function (s) {
    const isVowel = {
        a: 1,
        e: 1,
        i: 1,
        o: 1,
        u: 1,
        A: 1,
        E: 1,
        I: 1,
        O: 1,
        U: 1,
    }
    let low = 0, high = s.length - 1;
    let res = s.split('');
    while (low < high) {
        if (isVowel[res[low]] && isVowel[res[high]]) {
            //! Swapping 
            [res[low], res[high]] = [res[high], res[low]];
            low++, high--;
        } else if (!isVowel[s[low]]) low++;
        else high--;
    }
    return res.join('');
};
let s = "hello";
console.log(reverseVowels(s));
